src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,267 @@
+        {
+            this.Flush();
+            this.outputStream.Write(parentheses, 0, 1);
+        }
+
+        public void WritePaddingFunctionName(string functionName)
+        {
+            this.Flush();
+            this.outputStream.Write(encoding.GetBytes(functionName));
+        }
+
+        public void EndPaddingFunctionScope()
+        {
+            this.Flush();
+            this.outputStream.Write(parentheses, 1, 1);
+        }
+
+        public void StartObjectScope()
+        {
+            this.writer.WriteStartObject();
+        }
+
+        public void EndObjectScope()
+        {
+            this.writer.WriteEndObject();
+        }
+
+        public void StartArrayScope()
+        {
+            this.writer.WriteStartArray();
+        }
+
+        public void EndArrayScope()
+        {
+            this.writer.WriteEndArray();
+        }
+
+        public void WriteName(string name)
+        {
+            this.writer.WritePropertyName(name);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(bool value)
+        {
+            this.writer.WriteBooleanValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(int value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(float value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(short value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(long value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+            
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(double value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Guid value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(decimal value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(DateTimeOffset value)
+        {
+            this.writer.WriteStringValue(value);

```

Comment:
Curious if we should worry about culture here...

Reply:
This writes the date in ISO8601 date time format, the same as the existing `JsonWriter`. I don't think the format is culture dependent.